### PR TITLE
Import account to keycard

### LINKED ISSUE #7216: Import account to Keycard
# Problem

When creating new Status account, It should be possible to import existing account to Keycard by entering mnemonic phrase.

# Implementation

https://www.figma.com/file/UfQjpWl1hmRchHIyY3Wvu2nW/Hardwallet-light-flows?node-id=0%3A1

<blockquote><div> Figma</div><div><strong><a href="https://www.figma.com/file/UfQjpWl1hmRchHIyY3Wvu2nW/Hardwallet-light-flows">Figma</a></strong></div><div>Created with Figma</div></blockquote>

### PR DESCRIPTION
**Pull request by dmitryn**
fixes #7216 

### Summary:

Allow user to import account to keycard by entering passphrase.
Process is similar to keycard account creation. The only difference is that instead of generating mnemonic we're asking user to enter mnemonic.

#### Platforms (optional)
- Android

status: ready

### CODE REVIEW

File: src/status_im/hardwallet/core.cljs [hunk 1]
```diff
@@ -148,11 +150,27 @@
...
+    (fx/merge cofx
+              {:db (assoc-in db [:hardwallet :flow] :create)}
+              (navigation/navigate-to-cofx :hardwallet-authentication-method nil))
     (accounts.create/navigate-to-create-account-screen cofx)))
 
+(fx/defn navigate-to-recover-method
+  [{:keys [db] :as cofx}]
+  (if (hardwallet-supported? cofx)
+    (fx/merge cofx
+              {:db (assoc-in db [:hardwallet :flow] :import)}
```
Comment (goranjovic) [inline src/status_im/hardwallet/core.cljs:167]: Any reason why this `:flow` parameter couldn't be passed as a navigation parameter, e.g. 

`(navigation/navigate-to-cofx :hardwallet-authentication-method {:flow :import})`

and then get it with `:get-screen-params` subs.

**Review by goranjovic — APPROVED** [review]

File: src/status_im/hardwallet/core.cljs (hunk 1 above)
Comment (dmitryn) [reply src/status_im/hardwallet/core.cljs:167]: @goranjovic `:flow` parameter is used in several different screens. I didn't use `get-screen-params`, but seems like parameters there are related to one particular screen if i get it right. In my case, `:flow` is more like global parameter which is used throughout many screens.

Comment (goranjovic) [reply src/status_im/hardwallet/core.cljs:167]: Fair enough

File: src/status_im/ui/screens/hardwallet/components.cljs [hunk 2]
```diff
@@ -49,7 +49,7 @@
                                    :number    4
                                    :next-step :recovery-phrase}
    :card-ready                    {:label     :t/pairing
-                                   :number    5
+                                   :number    4
```
Comment (flexsurfer) [inline src/status_im/ui/screens/hardwallet/components.cljs:52]: just wondering what is this number for?

File: src/status_im/ui/screens/hardwallet/setup/views.cljs [hunk 3]
```diff
@@ -161,6 +166,48 @@
...
+  (letsubs [width [:dimensions/window-width]
+            recovered-account [:get-recover-account]]
+    (let [{:keys [passphrase-valid? passphrase-error passphrase-warning]} recovered-account
+          disabled? (not passphrase-valid?)]
+      [react/view styles/enter-pair-code-container
+       [react/view styles/enter-pair-code-title-container
+        [react/view
+         [react/text {:style           styles/center-title-text
+                      :number-of-lines 2
+                      :font            :bold}
```
Comment (flexsurfer) [inline src/status_im/ui/screens/hardwallet/setup/views.cljs]: hm do we have this font ? i thought it should be font-weight in style

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/ui/screens/hardwallet/setup/views.cljs (hunk 3 above)
Comment (dmitryn) [reply src/status_im/ui/screens/hardwallet/setup/views.cljs]: good point, removing this line.

File: src/status_im/ui/screens/hardwallet/components.cljs (hunk 2 above)
Comment (dmitryn) [reply src/status_im/ui/screens/hardwallet/components.cljs:52]: this just a number to display along with step name in topbar component:

screenshot 2019-02-27 19 47 18

**Comment by dmitryn** [comment]
@asemiankevich PR is ready for testing, here is android build:

http://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190227-180956-7122af-pr.apk

<blockquote></blockquote>

**Comment by asemiankevich** [comment]
@dmitryn is it possible to have mnemonic phrase validation as we have for our standart recovery flow? Spell checking, missing words, word count etc. @goranjovic made this pretty well and i think could assist if questions  occur.

also when trying to unpair the card -> error is shown

image

**Comment by dmitryn** [comment]
@asemiankevich added passphrase validation from standard recovery flow, please check when build will be ready.

> also when trying to unpair the card -> error is shown

Cannot reproduce. Maybe you did some extra steps? My flow was - import account (login from success screen), go to keycard settings, unpair, enter pin.

**Comment by asemiankevich** [comment]
@dmitryn yeah same flow to me , but can't proceed with PIN. Weird. I will try again with new build and let you know

**Comment by dmitryn** [comment]
@asemiankevich please check new build http://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190228-163411-84bb06-pr.apk

<blockquote></blockquote>

**Comment by asemiankevich** [comment]
looks great, thanks @dmitryn !

so, we have here:
- validation for pass phrase (same across the app)
- error dialog if user tries to recover keycard account with password
image
- error shown when user tries to recover existing keycard account (account already exists in status)
image

I believe we are good to go with this one, lets merge

**Pull request merged**